deposit/src/main/java/edu/unc/lib/deposit/SpringDepositServiceInitiator.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2008 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.deposit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+
+import edu.unc.lib.deposit.work.DepositSupervisor;
+
+/**
+ * Detects when the applications context has full initialized and then kicks off the deposit supervisor.
+ *
+ * @author bbpennel
+ *
+ */
+public class SpringDepositServiceInitiator implements ApplicationListener<ContextRefreshedEvent> {
+    private static final Logger log = LoggerFactory.getLogger(SpringDepositServiceInitiator.class);
+
+    private DepositSupervisor depositSupervisor;
+
+    @Override
+    public void onApplicationEvent(ContextRefreshedEvent event) {
+        log.info("Context initialized, starting deposit service");
```

Comment:
Terminology seems a bit weird since its called ContextRefreshedEvent, but this is what i found for how to get an event for when the context has finished loading